tests/test_xml_parser_main_metadata_together.c

```diff
@@ -262,37 +263,232 @@ test_cr_xml_parse_main_metadata_together_08_long_primary(void)
+//     int ret = cr_xml_parse_main_metadata_together(TEST_REPO_02_PRIMARY, TEST_REPO_02_FILELISTS, TEST_REPO_02_OTHER,
+//                                                   NULL, NULL, pkgcb_interrupt, &parsed, NULL, NULL, TRUE, &tmp_err);
+//     g_assert(tmp_err != NULL);
+//     g_error_free(tmp_err);
+//     g_assert_cmpint(ret, ==, CRE_CBINTERRUPTED);
+//     g_assert_cmpint(parsed, ==, 1);
+// }
+
+// static void
+// test_cr_xml_parse_main_metadata_together_06_warnings_bad_file_type(void)
+// {
+//     int parsed = 0;
+//     char *warnmsgs;
+//     GError *tmp_err = NULL;
+//     GString *warn_strings = g_string_new(0);
+//     int ret = cr_xml_parse_main_metadata_together(TEST_REPO_02_PRIMARY, TEST_MRF_BAD_TYPE_FIL, TEST_REPO_02_OTHER,
+//                                                   NULL, NULL, pkgcb, &parsed, warningcb, warn_strings, TRUE, &tmp_err);
+//     g_assert(tmp_err == NULL);
+//     g_assert_cmpint(ret, ==, CRE_OK);
+//     g_assert_cmpint(parsed, ==, 2);
+//     warnmsgs = g_string_free(warn_strings, FALSE);
+//     g_assert_cmpstr(warnmsgs, ==, "Unknown file type \"foo\";");
+//     g_free(warnmsgs);
+// }
+
+static void
+test_cr_xml_package_iterator_07_warningcb_interrupt(void)
+{
+    int parsed = 0;
+    int numofwarnings = 0;
+    GError *tmp_err = NULL;
+    cr_Package *package = NULL;
+
+    cr_PkgIterator *pkg_iterator = cr_PkgIterator_new(
+        TEST_REPO_02_PRIMARY, TEST_MRF_BAD_TYPE_FIL, TEST_REPO_02_OTHER, NULL, NULL, warningcb_interrupt, &numofwarnings, TRUE, &tmp_err);
+
+    while ((package = cr_PkgIterator_parse_next(pkg_iterator, &tmp_err))) {
+        parsed++;
+        cr_package_free(package);
+    }
+
+    g_assert(tmp_err != NULL);
+    g_assert_cmpint(parsed, ==, 0);
+    g_assert_cmpint(tmp_err->code, ==, CRE_CBINTERRUPTED);
+    g_assert_cmpint(numofwarnings, ==, 1);
+    g_clear_error(&tmp_err);
+}
+
+static void
+test_cr_xml_package_iterator_08_long_primary(void)
+{
+    int parsed = 0;
+    GError *tmp_err = NULL;
+    cr_Package *package = NULL;
+
+    cr_PkgIterator *pkg_iterator = cr_PkgIterator_new(
+        TEST_LONG_PRIMARY, TEST_REPO_02_FILELISTS, TEST_REPO_02_OTHER, NULL, NULL, NULL, NULL, TRUE, &tmp_err);
+
+    while ((package = cr_PkgIterator_parse_next(pkg_iterator, &tmp_err))) {
+        parsed++;
+        cr_package_free(package);
+    }
+
+    g_assert(cr_PkgIterator_is_finished(pkg_iterator));
+    cr_PkgIterator_free(pkg_iterator, &tmp_err);
+
+    g_assert(tmp_err == NULL);
+    g_assert_cmpint(parsed, ==, 2);
+}
+
 int
 main(int argc, char *argv[])
 {
     g_test_init(&argc, &argv, NULL);
 
-    g_test_add_func("/xml_parser_filelists/test_cr_xml_parse_main_metadata_together_00",
+    g_test_add_func("/xml_parser_main_metadata/test_cr_xml_parse_main_metadata_together_00",
                     test_cr_xml_parse_main_metadata_together_00);
 
-    g_test_add_func("/xml_parser_filelists/test_cr_xml_parse_main_metadata_together_01_out_of_order_pkgs",
+    // LEAKS
```

Comment:
These are all passing, but a few of them do leak memory atm.

Reply:
Interesting some of them are failing for me without the changes I just suggested above.
Are you sure you are testing with the modified library and not the one present on the system? Or given that all of my fails are caused by `asserts` perhaps those could somehow behave differently?